cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/mapper/SdkClientExceptionMapper.java

```diff
@@ -16,14 +18,34 @@
     @Inject
     private AwsEncodedAuthorizationFailureMessageDecoder awsEncodedAuthorizationFailureMessageDecoder;
 
-    public RuntimeException map(AwsCredentialView awsCredentialView, String region, SdkClientException e) {
+    public RuntimeException map(AwsCredentialView awsCredentialView, String region, SdkClientException e, Signature signature) {
         String message = awsEncodedAuthorizationFailureMessageDecoder.decodeAuthorizationFailureMessageIfNeeded(awsCredentialView, region, e.getMessage());
+        String methodName = signature.getName();
+        String pre = "Cannot execute method: " + methodName + ". ";
         if (!message.equals(e.getMessage())) {
+            message = addMethodNameIfNotContains(message, methodName, pre);
             return new CloudConnectorException(message, e);
         }
         if (message.contains("Rate exceeded") || message.contains("Request limit exceeded")) {
+            message = addMethodNameIfNotContains(message, methodName, pre);
             return new ActionFailedException(message);
         }
+        extendErrorMessageWithMethodNameIfAmazonServiceException(e, methodName, pre);
         return e;
     }
+
+    private String addMethodNameIfNotContains(String message, String methodName, String pre) {
+        if (!message.toLowerCase().contains(methodName.toLowerCase())) {
+            message = pre + message;
+        }
+        return message;
+    }
+
+    // We use the AmazonServiceException to check the cloudformation exists or not. And maybe we built much more logic to this exception.
+    // Therefore the error messages are updated instead of the wrap
+    private void extendErrorMessageWithMethodNameIfAmazonServiceException(SdkClientException e, String methodName, String pre) {
+        if (e instanceof AmazonServiceException && !e.getMessage().toLowerCase().contains(methodName.toLowerCase())) {
```

Comment:
This could be a bit simpler if you move the if instanceof to the map() method, and modify the e parameter type here to AmazonServiceException. It would also better match the previous conditionals.